internal/revision/database/revision.go

```diff
@@ -0,0 +1,292 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package database contains the management of interactions with the database
+// for createion and storage of the wrapped keys that encrypet revision certificates.
+package database
+
+import (
+	"context"
+	"crypto/rand"
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/google/exposure-notifications-server/internal/database"
+	"github.com/google/exposure-notifications-server/internal/logging"
+	"github.com/google/exposure-notifications-server/pkg/keys"
+	"github.com/jackc/pgx/v4"
+)
+
+// RevisionKey represents an individual revision key.
+type RevisionKey struct {
+	KeyID         int64
+	AAD           []byte // AAD for the wrapping/unwrapping of the cipher block.
+	WrappedCipher []byte
+	CreatedAt     time.Time
+	Allowed       bool
+
+	// The unwrapped cipher.
+	DEK []byte
+}
+
+// KeyIDString returns the keyID as a string that can be used in the encoded revision tokens.
+func (r *RevisionKey) KeyIDString() string {
+	return fmt.Sprintf("%d", r.KeyID)
```

Comment:
If we're already printing it as a string, why not keep it as a string in the proto?

Also strconv.Itoa(r.KeyID) is much faster.